packages/block-editor/src/components/border-radius-control/index.js

```diff
@@ -49,7 +50,9 @@ export default function BorderRadiusControl( { onChange, values } ) {
 		! hasDefinedValues( values ) || ! hasMixedValues( values )
 	);
 
-	const units = useCustomUnits( { availableUnits: [ 'px', 'em', 'rem' ] } );
+	const units = useCustomUnits( {
+		availableUnits: useSetting( 'border.units' ) || [ 'px', 'em', 'rem' ],
```

Comment:
In https://github.com/WordPress/gutenberg/pull/33280#discussion_r666068909 we consolidated all units into `spacing.units` so there's no longer a `layout.units`. I wonder if we can do the same here.

See this specific thread for more context https://github.com/WordPress/gutenberg/pull/33280#discussion_r666068909 The concerns were that the relation between both wasn't well understood and the implementation of `UnitControl` made layout values dependant on spacing values (layout couldn't have values that spacing also had).

Reply:
I can change this to `spacing.units` if desired. I'm not sure I follow why we wouldn't potentially want a different set of units for borders. 

For example, do `vh` and `vw` make sense for borders? Would a theme author wish to have a different set of available units for borders as opposed to spacing?

> (layout couldn't have values that spacing also had).

Unless I'm missing something, is it not that the supplied units had to be a subset of spacing units? That makes sense to me for borders. If you didn't want `em` or `rem` for spacing, I can see them also not being desired for border units.

From the linked thread, you mention there that units can only be a subset of spacing.units.

> This means that what you pass as units can only be a subset of spacing.units

I'm also fine with a desire to simplify the handling and config of units. If the extra flexibility of different border units isn't worth the cost, I'll change this PR accordingly.

Reply:
After some further testing, the borders behave ok with all units. I also found I'd missed the global styles' border panel units. 

Given that, I've taken the opportunity to switch everything to use `spacing.units` while updating the global styles panel.